pkg/policy/api/cidr_test.go

```diff
@@ -154,6 +154,110 @@ func TestGetAsEndpointSelectors(t *testing.T) {
+			out = append(out, sel)
+		}
+		return out
+	}
+
+	tt := []struct {
+		name                   string
+		rule                   CIDRRule
+		expected               EndpointSelectorSlice
+		enableIPv4, enableIPv6 bool
+	}{
+		{
+			name:       "basic cidr",
+			rule:       CIDRRule{Cidr: "1.2.3.4/32"},
+			expected:   selectors("cidr.1.2.3.4/32"),
+			enableIPv4: true, enableIPv6: true,
+		},
+		{
+			name: "except",
+			rule: CIDRRule{Cidr: "1.0.0.0/8", ExceptCIDRs: []CIDR{"1.2.3.4/32"}},
+			expected: EndpointSelectorSlice{NewESFromMatchRequirements(nil, []v1.LabelSelectorRequirement{
+				{Key: "cidr.1.0.0.0/8", Operator: "Exists"},
+				{Key: "cidr.1.2.3.4/32", Operator: "DoesNotExist"},
+			})},
+			enableIPv4: true, enableIPv6: true,
+		},
+
+		{
+			name:       "cidr group",
+			rule:       CIDRRule{CIDRGroupRef: "foo"},
+			expected:   selectors("cidrgroup.io.cilium.policy.cidrgroupname/foo"),
+			enableIPv4: true, enableIPv6: true,
+		},
+		{
+			name: "cidr group ls",
+			rule: CIDRRule{CIDRGroupSelector: &v1.LabelSelector{
+				MatchLabels: map[string]v1.MatchLabelsValue{"foo": "bar"},
+			}},
+			expected:   EndpointSelectorSlice{NewESFromLabels(labels.NewLabel("foo", "bar", "cidrgroup"))},
+			enableIPv4: true, enableIPv6: true,
+		},
+
+		{
+			name:       "world v4 ss",
+			rule:       CIDRRule{Cidr: "0.0.0.0/0"},
+			expected:   selectors("cidr.0.0.0.0/0", "reserved.world"),
+			enableIPv4: true, enableIPv6: false,
+		},
+		{
+			name:       "world v4 ds",
+			rule:       CIDRRule{Cidr: "0.0.0.0/0"},
+			expected:   selectors("cidr.0.0.0.0/0", "reserved.world-ipv4"),
+			enableIPv4: true, enableIPv6: true,
+		},
+		{
+			name:       "world v6 ss",
+			rule:       CIDRRule{Cidr: "::/0"},
+			expected:   selectors("cidr.0--0/0", "reserved.world"),
+			enableIPv4: false, enableIPv6: true,
+		},
+		{
+			name:       "world v6 ds",
+			rule:       CIDRRule{Cidr: "::/0"},
+			expected:   selectors("cidr.0--0/0", "reserved.world-ipv6"),
+			enableIPv4: true, enableIPv6: true,
+		},
+
+		{
+			name: "world v4 ds except",
+			rule: CIDRRule{Cidr: "0.0.0.0/0", ExceptCIDRs: []CIDR{"1.2.3.4/32"}},
+			expected: EndpointSelectorSlice{
+				NewESFromMatchRequirements(nil, []v1.LabelSelectorRequirement{
+					{Key: "cidr.0.0.0.0/0", Operator: "Exists"},
+					{Key: "cidr.1.2.3.4/32", Operator: "DoesNotExist"},
+				}),
+				NewESFromMatchRequirements(nil, []v1.LabelSelectorRequirement{
+					{Key: "reserved.world-ipv4", Operator: "Exists"},
+					{Key: "cidr.1.2.3.4/32", Operator: "DoesNotExist"},
+				}),
+			},
+			enableIPv4: true, enableIPv6: true,
+		},
+	}
+
+	for _, test := range tt {
+		t.Run(test.name, func(t *testing.T) {
+			option.Config.EnableIPv6 = test.enableIPv6
+			option.Config.EnableIPv4 = test.enableIPv4
+			result := (CIDRRuleSlice{test.rule}).GetAsEndpointSelectors()
+			require.Equal(t, test.expected, result)
+		})
+	}
+	option.Config.EnableIPv4 = true
+	option.Config.EnableIPv6 = true
```

Comment:
We should probably return these configs back to their original value after the test, so that we aren't polluting other tests. May not cause problems now, but in the future it could and it'll be very confusing to that developer